catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/AccessControlPreIngestPlugin.java

```diff
@@ -30,99 +32,86 @@
 import org.apache.commons.lang.StringUtils;
 import org.apache.shiro.SecurityUtils;
 import org.apache.shiro.subject.Subject;
-import org.codice.ddf.catalog.ui.sharing.ShareableMetacardImpl;
 
-public class ShareableMetacardPreIngestPlugin implements PreIngestPlugin {
+public class AccessControlPreIngestPlugin implements PreIngestPlugin {
 
   private final SubjectIdentity subjectIdentity;
 
-  public ShareableMetacardPreIngestPlugin(SubjectIdentity subjectIdentity) {
+  public AccessControlPreIngestPlugin(SubjectIdentity subjectIdentity) {
     this.subjectIdentity = subjectIdentity;
   }
 
-  private static Map<String, ShareableMetacardImpl> getPreviousShareableMetacards(
-      UpdateRequest request) {
+  private static Map<String, Metacard> getPreviousMetacards(UpdateRequest request) {
     OperationTransaction operationTransaction =
         (OperationTransaction) request.getProperties().get(Constants.OPERATION_TRANSACTION_KEY);
 
     return operationTransaction
         .getPreviousStateMetacards()
         .stream()
-        .filter(ShareableMetacardImpl::canShare)
-        .map(ShareableMetacardImpl::createOrThrow)
+        .filter(containsACLAttributes)
         .collect(Collectors.toMap(Metacard::getId, m -> m));
   }
 
-  private static void copyOwner(ShareableMetacardImpl source, ShareableMetacardImpl target) {
+  private static void copyOwner(Metacard source, Metacard target) {
     if (source == null || target == null) {
       return;
     }
-    target.setOwner(source.getOwner());
+    AccessControlUtil.setOwner(target, AccessControlUtil.getOwner(source));
   }
 
   /**
    * Ensures a sharing metacard has an owner.
    *
    * @param request the {@link CreateRequest} to process
-   * @return
    * @throws PluginExecutionException
    * @throws StopProcessingException - if the current subject doesn't have an email attribute
```

Comment:
✏️  Remove PluginExectionException throw and update StopProcessingException statement by replacing "email attribute" with "owner attribute"